force-app/main/default/classes/FieldStatisticsREST.cls

```diff
@@ -0,0 +1,38 @@
+@RestResource(urlMapping='/api/dbstatistics')
+global with sharing class FieldStatisticsREST {
+    @HttpGet
+    global static Map<String, Integer> getFieldStatistics(){
```

Comment:
I suggest calculating field usage using Batch, as with the current implementation getFieldStatistics() will be able to query up to  50000 records and calculate statistics for them only